meteomatics/tests/integration_test.rs

```diff
@@ -0,0 +1,785 @@
+    let interval = Duration::hours(12);
+
+    // Create Parameters
+    let parameter =String::from("t_2m:C");
+
+    // Create Location
+    let bbox: BBox = BBox {
+        lat_min: 52.40,
+        lat_max: 52.50,
+        lon_min: 13.40,
+        lon_max: 13.50,
+        lat_res: 0.05,
+        lon_res: 0.05
+    };
+
+    // Create file name
+    let file_name: String = String::from("tests/netcdf/my_netcdf.nc");
+
+    // Call endpoint
+    meteomatics_connector
+        .query_netcdf(
+            &start_date, &end_date, &interval, &parameter, &bbox, &file_name, &None
+        )
+        .await
+        .unwrap();
+
+    // Make some tests
+    let file = netcdf::open(&file_name).unwrap();
+    let var = &file.variable("t_2m").expect("Could not find variable 't_2m");
+
+    // Check value: ds_rust["t_2m"].data[0,0,0]
+    let xr_test: f64 = 6.81058931350708;
+    let temp_f64: f64 = var.value(Some(&[0,0,0])).unwrap();
+    assert_eq!(xr_test, temp_f64);
+
+    // Check another value: ds_rust["t_2m"].data[2,2,2]
+    let xr_test: f64 = 5.269172668457031;
+    let temp_f64: f64 = var.value(Some(&[2,2,2])).unwrap();
+    assert_eq!(xr_test, temp_f64);
+
+    // Remove the file    
+    let dir: &Path = Path::new(&file_name).parent().unwrap();
+    fs::remove_file(&file_name).unwrap();
+    fs::remove_dir_all(&dir).unwrap();
+}
+
+#[tokio::test]
+// TODO: Add more tests for actual data content.
+async fn query_png() {
+    // Query using rust connector
+    // Credentials
+    dotenv().ok();
+    let api_key: String = env::var("METEOMATICS_PW").unwrap();
+    let api_user: String = env::var("METEOMATICS_USER").unwrap();
+    
+    // Create API connector
+    let meteomatics_connector = APIClient::new(
+        api_user,
+        api_key,
+        10,
+    );
+
+    // Create time information
+    // 1989-11-09 19:00:00 --> 18:00:00 UTC
+    let start_date = Utc.ymd(1989, 11, 9).and_hms_micro(18, 0, 0, 0);
+
+    // Create Parameters
+    let parameter =String::from("t_2m:C");
+
+    // Create Location
+    let bbox: BBox = BBox {
+        lat_min: 45.8179716,
+        lat_max: 47.8084648,
+        lon_min: 5.9559113,
+        lon_max: 10.4922941,
+        lat_res: 0.01,
+        lon_res: 0.01
+    };
+
+    // Create file name
+    let file_name: String = String::from("tests/png/my_png.png");
+
+    // Call endpoint
+    meteomatics_connector
+        .query_grid_png(
+            &start_date, &parameter, &bbox, &file_name, &None
+        )
+        .await
+        .unwrap();
+
+    let decoder = png::Decoder::new(fs::File::open(&file_name).unwrap());
+    let reader = decoder.read_info().unwrap();
+
+    // Inspect more details of the last read frame.
+    assert_eq!(454, reader.info().width);
+    assert_eq!(200, reader.info().height);
+    
+    // Remove the file    
+    let dir: &Path = Path::new(&file_name).parent().unwrap();
+    fs::remove_file(&file_name).unwrap();
+    fs::remove_dir_all(&dir).unwrap();
```

Comment:
Maybe at the end some assertin that the file is actually removed, like checking for existence and it should return false. 
assert!(!path_exists)

Reply:
Good idea, thanks!